src/btree/bt_delete.c

```diff
@@ -173,7 +176,7 @@ __wt_delete_page_rollback(WT_SESSION_IMPL *session, WT_REF *ref)
 			 */
 			if (__wt_atomic_casv32(
 			    &ref->state, WT_REF_DELETED, WT_REF_DISK))
-				return;
+				to_return = true;
```

Comment:
It would be more inline with our style to use a `goto done` here, and add a `done` label at the end of the loop. See [bt_walk.c](https://github.com/wiredtiger/wiredtiger/blob/develop/src/btree/bt_walk.c#L400) for an example